util/data_parser.py

```diff
@@ -41,13 +51,43 @@ def get_ratings():
         config = DataParser.get_config()
         cursor.execute("use %s" % config["database"]["database_name"])
         cursor.execute("set group_concat_max_len=100000")
-        cursor.execute("select user_id, concat('[', group_concat(article_id separator ', '),']') from articles_users group by user_id")
+        cursor.execute("select user_id, group_concat(article_id separator ', ') from articles_users group by user_id")
         ratings_hash = {}
         for (user_id, json) in cursor:
-            ratings_hash[int(user_id)] = json
+            ratings_hash[int(user_id)] = DataParser.listify(json)
         DataParser.clean_up(db, cursor)
         return ratings_hash
 
+   
+    @staticmethod
+    def get_row_count(table_name):
+        """
+        @returns int indicating number of users
+        """
+        db = DataParser.get_connection()
+        cursor = db.cursor()
+        config = DataParser.get_config()
+        cursor.execute("use %s" % config["database"]["database_name"])
+        cursor.execute("select count(*) as c from %s" % table_name)
+        row = cursor.fetchone()
+        return int(row[0])
+
+    @staticmethod
+    def get_ratings_matrix():
+        """
+        @returns A matrix with between users and documents. 1 indicates that the user has the document
+        in his library, 0 otherwise.
+        """
+        ratings_hash = DataParser.get_ratings_hash()
+        num_users = DataParser.get_row_count("users")
+        num_articles = DataParser.get_row_count("articles")
+        ratings_matrix = [[0] * num_articles for _ in range(num_users)]
```

Comment:
I thought we are going to use numpy matrices? or what do you think, it could also have some drawbacks